lib/matplotlib/ticker.py

```diff
@@ -1292,6 +1295,117 @@ def convert_to_pct(self, x):
 
 
+class TransformFormatter(Formatter):
+    """
+    A generalized alternative to `FuncFormatter` that allows the tick
+    values to be transformed arbitrarily before being passed off to
+    the formatting function.
+
+    This class accepts a callable transform and a callable formatter.
+    The transform function may return any type that is acceptable to the
+    formatter, not necessarily just a `float`. For example, using
+    ``int`` as the transform allows integer format strings such as
+    `'{x:d}'` to be used with :class:`StrMethodFormatter`.
+
+    The formatter can be either a :class:`matplotlib.ticker.Formatter`
+    or any other callable with the signature ``formatter(x, pos=None)``.
+    If the formatter is an actual formatter, most methods of this class
+    will delegate directly to it. A notable exception is `set_locs`,
+    which will apply the transformation to each of the passed in
+    locations before delegating. In the case of a generic callable, this
+    class will handle all of the ``Formatter`` functionality directly.
+
+    .. note::
+
+        Instantiating this class with the special case of
+        ``transform=lambda x: x`` makes it exactly equivalent to
+        :class:`matplotlib.ticker.FuncFormatter`, unless `formatter` is
+        set to a ``Formatter`` instance instead of a simple callable.
+    """
+    def __init__(self, transform, formatter=ScalarFormatter()):
+        self.transform = transform
+        self.set_formatter(formatter)
+
+    def set_formatter(self, formatter):
+        """
+        Changes the underlying formatter used to actually format the
+        transformed values.
+
+        The input may be an instance of
+        :class:`matplotlib.ticker.Formatter` or an other callable with
+        the same signature.
+        """
+        self.formatter = formatter
+        self._need_redirect = isinstance(formatter, Formatter)
+        if self.locs is not None:
+            self.set_locs(self.locs)
+        # This will implicitly set the view and data interval for the
+        # underlying transform correctly.
+        self.set_axis(self.axis)
+
+    def __call__(self, x, pos=None):
+        return self.formatter(self.transform(x), pos)
+
+    def _invoke_redirect(self, name, both, *args, **kwargs):
+        """
+        Invokes the specified method on the underlying formatter if
+        possible, or on `self` if not. If `both` is `True`, the
+        formatter's return value if returned if possible. This method
+        allows the actual formatter to be a generic callable rather than
+        a Formatter.
+        """
+        if self._need_redirect:
+            retval = getattr(self.formatter, name)(*args, **kwargs)
+            if not both:
+                return retval
+        # only evaluate this if necessary
+        default = getattr(super(TransformFormatter, self), name)
+        retval2 = default(*args, **kwargs)
+        return retval if self._need_redirect else retval2
```

Comment:
Huh, I am surprised that this does not raise a `NameError` but it makes sense that it does not.